imperium/shipyard.py

```diff
@@ -0,0 +1,136 @@
+            base_stats_layout.addWidget(new_line_edit, row, 2)
+
+            return new_line_edit
+
+        # Tonnage
+        self.tonnage_line_edit = add_stat_to_layout("Tonnage:", 0, signal_function=self.edit_tonnage, force_int=True)
+
+        # Cargo
+        self.cargo_line_edit = add_stat_to_layout("Cargo:", 1, read_only=True)
+
+        # Fuel
+        self.fuel_line_edit = add_stat_to_layout("Fuel:", 2, signal_function=self.edit_fuel, force_int=True)
+
+        # Jump
+        self.jump_line_edit = add_stat_to_layout("Jump:", 3, read_only=True)
+
+        # Thrust
+        self.thrust_line_edit = add_stat_to_layout("Thrust:", 4, read_only=True)
+
+        # Hull HP
+        self.hull_hp_line_edit = add_stat_to_layout("Hull HP:", 5, read_only=True)
+
+        # Structure HP
+        self.structure_hp_line_edit = add_stat_to_layout("Structure HP:", 6, read_only=True)
+
+        # Armor
+        self.armour_line_edit = add_stat_to_layout("Armour:", 7, read_only=True)
+
+        # Cost
+        self.cost_line_edit = add_stat_to_layout("Cost:", 8, read_only=True)
+        
+        # Grid layout
+        base_stats_group.setLayout(base_stats_layout)
+        ###################################
+        ###  END: Base Stats Grid       ###
+        ###################################
+
+        # Overall layout grid
+        layout = QGridLayout()
+        layout.addWidget(base_stats_group, 0, 0)
+        self.setLayout(layout)
+
+        # Update to current stats
+        self.update_stats()
+
+    def update_stats(self):
+        """
+        Updates the UI with the current Spacecraft stats
+        """
+        self.tonnage_line_edit.setText(str(      self.spacecraft.tonnage            ))
+        self.cargo_line_edit.setText(str(        self.spacecraft.cargo              ))
+        self.fuel_line_edit.setText(str(         self.spacecraft.fuel_max           ))
+        self.jump_line_edit.setText(str(         self.spacecraft.jump               ))
+        self.thrust_line_edit.setText(str(       self.spacecraft.thrust             ))
+        self.hull_hp_line_edit.setText(str(      self.spacecraft.hull_hp            ))
+        self.structure_hp_line_edit.setText(str( self.spacecraft.structure_hp       ))
+        self.armour_line_edit.setText(str(       self.spacecraft.get_armor_rating() ))
+        self.cost_line_edit.setText(str(         self.spacecraft.cost_total         ))
+
+    def edit_tonnage(self):
+        """
+        Update the spacecraft tonnage
+        """
+        new_tonnage = int(self.tonnage_line_edit.text())
+        self.spacecraft.set_tonnage(new_tonnage)
+
+    def edit_fuel(self):
+        """
+        Update the spacecraft max fuel
+        """
+        new_fuel = int(self.fuel_line_edit.text())
+        self.spacecraft.set_fuel(new_fuel)
+
+
+if __name__ == '__main__':
+    import sys
+
+    app = QApplication(sys.argv)
+    window = Window()
+    window.show()
+    sys.exit(app.exec_())
```

Comment:
Newline here, por favor

Reply:
This has newline, will show red return arrow if not (I think I've not realized this in the past as well)

Can verify by clicking the edit button for the file.